shared/src/types/types.ts

```diff
@@ -27,16 +27,31 @@ export interface Violation {
   effort?: number;
 }
 
-export type ViolationWithID = Violation & {
+export type EnhancedViolation = Violation & {
   id: string;
+  rulesetName?: string;
+  violationName?: string;
 };
 
+export interface EnhancedIncident extends Incident {
```

Comment:
Should this just be one type Incident? do we ever want non enhance incidents?

Reply:
I think this structure comes into play when we unmarshal the analysis response from kai. I'm not certain if only having the enhanced would cause any problems.

I rate my conviction on this...meh...but I would prefer to keep them distinct with a comment that Incident is what we get from analysis and EnhancedIncident is what we send to kai.

Reply:
The enhanced incidents seems the IDE version of https://github.com/konveyor/kai/blob/82e195916be14eddd08c4e2bfb69afc0880edfcb/kai/analyzer_types.py#L89-L106

@ibolton336  please add a comment pointing to that type to keep both aligned in the future

Reply:
@JonahSussman please take a look if the flattening in the IDE works the same as in the backend.